scrunch/datasets.py

```diff
@@ -680,7 +680,51 @@ def __readonly__(self, *args, **kwargs):
     del __readonly__
 
 
-class Dataset(object):
+class DatasetVariablesMixin(collections.Mapping):
+    """
+    Handles dataset variable iteration
+    """
+
+    def __init__(self, ds_resource):
+        self.ds_resource = ds_resource
+        self._vars = self.ds_resource.variables.index.items()
```

Comment:
Given the comments below, this assignation should be inside a method `self._reload_variables()` that gets called here and also after each variable creation.

We should also store the `.by('alias')` there so we don't need to fetch the variable catalog each time we want to access a variable.